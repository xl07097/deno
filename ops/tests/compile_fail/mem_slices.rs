// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.

use deno_ops::op;

#[op]
fn sync_test(slice: &mut [u32]) {
  //
}

#[op]
async fn async_test(slice: &[u8]) {
  // Memory slices are not allowed in async ops.
}

#[op]
fn async_test2(slice: &mut [u8]) -> impl Future<Output = ()> {
  // Memory slices are not allowed in async ops, even when not implemented as an
  // async function.
  async {}
}

fn main() {
  // pass
}
