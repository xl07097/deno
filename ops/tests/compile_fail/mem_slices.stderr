error: custom attribute panicked
  --> tests/compile_fail/mem_slices.rs:10:1
   |
10 | #[op]
   | ^^^^^
   |
   = help: message: Memory slices are not allowed in async ops

error: custom attribute panicked
  --> tests/compile_fail/mem_slices.rs:15:1
   |
15 | #[op]
   | ^^^^^
   |
   = help: message: Memory slices are not allowed in async ops
